### PR TITLE
[Interview] Candidate: Pavan Sheshadri

### PR DESCRIPTION
**Pull request by ssankaranarayanan-fsp**

### CODE REVIEW

File: timeseries_toolbox/data_utils.py [hunk 1]
```diff
@@ -84,24 +84,33 @@ def load_data(
...
-    train_dataset, val_dataset = torch.utils.data.random_split(
-        dataset,
-        [train_size, val_size],
-    )
+    if shuffle:
+        train_dataset, val_dataset = torch.utils.data.random_split(
+            dataset,
+            [train_size, val_size],
+        )
+    else:
```
Comment (pseshadri-sf) [inline timeseries_toolbox/data_utils.py:108]: if-else logic is flipped. Currently if shuffle is False, no shuffling occurs, and if shuffle is True, shuffling occurs

File: timeseries_toolbox/models/gaussian_transformer.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+import torch
+from torch import Tensor
+import torch.nn as nn
+from torch.nn import functional as F
+from torch.optim import Optimizer
+
+from timeseries_toolbox.models.base_timeseries import BaseTimeSeriesModel
+
+
+class causal_self_attention(nn.Module):
```
Comment (pseshadri-sf) [inline timeseries_toolbox/models/gaussian_transformer.py:13]: use camel case in all class names

File: timeseries_toolbox/models/gaussian_transformer.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        attn /= math.sqrt(C)
+        attn = attn.masked_fill(self.causal_mask[:, :L, :L] == 0,
+                                float('-inf'))
+        attn = F.softmax(attn, dim=-1)
+        # Apply attention to values
+        net_out = attn @ value  # net_out shape (B, L, C)
+        return self.out_proj(net_out)  # Final output shape (B, L, C)
+
+
+class transformer_block(nn.Module):
```
Comment (pseshadri-sf) [inline timeseries_toolbox/models/gaussian_transformer.py:74]: Move TransformerBlock and all non-GT blocks to separate "model_blocks.py" file

File: timeseries_toolbox/models/gaussian_transformer.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+        Returns:
+            Output of shape (batch_size, seq_len, embedding_dim).
+        """
+        # B = batch_size, L = seq_len, C = embedding_dim
+        B, L, C = net_in.size()
+        # Get Q, K, V, each has shape (B, L, C)
+        query, key, value = self.qkv_proj(net_in).chunk(3, dim=-1)
+        # Compute attention scores
+        attn = (query @ key.transpose(-2, -1))  # Attn resulting shape (B, L, C)
```
Comment (pseshadri-sf) [inline timeseries_toolbox/models/gaussian_transformer.py:64]: fix transpose for correct matrix mul: B L C x B C L, currently does B L C x C B L

File: timeseries_toolbox/models/gaussian_transformer.py [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        pred_var = torch.relu(pred_var) + 1e-6
+        sq_err = (labels - pred_mean).square()
+        loss = 0.5 * (
+            # torch.log(torch.tensor(2 * torch.pi))
+            + torch.log(pred_var)
+            + sq_err / pred_var
+        ).mean()
+        return loss, {
+            "loss": loss.item(),
+            "nll": loss.item(),
```
Comment (pseshadri-sf) [inline timeseries_toolbox/models/gaussian_transformer.py:237]: should be gaussian likelihood
